### PR TITLE
Update sign-up page and user messages, hotfix sign-out

### LINKED ISSUE #94: Add features to the sign_in, sign_up pages
- Check that the email ends with @mta.ac.il
- Display successful / not successful registration and login messages 
- to be added

### PR DESCRIPTION
**Pull request by LeonRabinovich**
**The following changes were made:**

1. Fix a bug where a 404 error is displayed after sign-out - via any page which isn't the homepage. (app_layout.html)
2. Change the design of the user message which is displayed after sign-in/sign-out/sign-up (app_layout.html)
3. Add email validation to the sign-up form - email now must end with `@mta.ac.il `(forms.py, sign_up_tests.py)

- **New user message design:**

image

- **Sign-up form with the addition of academic email check:**

image

Fix: #94

### CODE REVIEW

File: homepage/tests/sign_up_tests.py [hunk 1]
```diff
@@ -10,20 +10,20 @@ class TestSignUp:
...
-            'username': 'valid_username', 'email': 'valid@email.com', 'password1': 'pw123123', 'password2': 'pw123123'}
+            'username': 'valid_username', 'email': 'valid@mta.ac.il', 'password1': 'pw123123', 'password2': 'pw123123'}
 
     @pytest.mark.parametrize("invalid_user_details", [
         # username cannot be None
-        {'username': None, 'email': 'valid@email.com', 'password1': 'pw123123', 'password2': 'pw123123'},
-        # email cannot be None
-        {'username': 'valid_username', None: 'valid@email.com', 'password1': 'pw123123', 'password2': 'pw123123'},
+        {'username': None, 'email': 'valid@mta.ac.il', 'password1': 'pw123123', 'password2': 'pw123123'},
+        # email must contain @mta.ac.il
```
Comment (VeraTs) [inline homepage/tests/sign_up_tests.py]: * must end with @mta.ac.il

**Review by VeraTs — COMMENTED** [review]
Looks great!
Just one small typing error

File: homepage/tests/sign_up_tests.py (hunk 1 above)
Comment (LeonRabinovich) [reply homepage/tests/sign_up_tests.py]: Fixed

**Review by oriAdler — APPROVED** [review]
LGTM

File: homepage/forms.py [hunk 2]
```diff
@@ -35,17 +35,26 @@ class FilterForm(forms.Form):
...
+    email = forms.EmailField(initial='@mta.ac.il', help_text='Required. must use an academic email.', required=True)
 
     class Meta:
         model = User
 
         # There are 2 password fields to confirm the password (pw1 and pw2 are built-in function names)
         fields = ("username", "email", "password1", "password2")
 
+    def clean_email(self):
+        ACADEMIC_EMAIL_SUB_STRING = '@mta.ac.il'
```
Comment (kobihk) [inline homepage/forms.py]:
```suggestion
        ACADEMIC_EMAIL_SUFFIX = '@mta.ac.il'

```

File: homepage/forms.py [hunk 3]
```diff
@@ -35,17 +35,26 @@ class FilterForm(forms.Form):
 
 # This class uses the built in django UserCreationForm and adds an email field
 class SignUpForm(UserCreationForm):
-    email = forms.EmailField(required=True)
+    email = forms.EmailField(initial='@mta.ac.il', help_text='Required. must use an academic email.', required=True)
```
Comment (kobihk) [inline homepage/forms.py:41]: need to think about other tests which use invalid mail as valid.
maybe the part of them in the list below:
homepage/migrations/0002_user_test_data.py:15:            ('testUser1', 'user1@gmail.com', 'password123'),
homepage/migrations/0002_user_test_data.py:16:            ('testUser2', 'David@gmail.com', 'Password456'),
homepage/migrations/0002_user_test_data.py:17:            ('testUser3', 'User3@gmail.com', 'User3Password')
homepage/tests/sign_in_tests.py:13:        User.objects.create_user(username='valid_username', email='valid@email.com', password='pw123123')
homepage/tests/sign_out_tests.py:10:        User.objects.create_user(username='valid_username', email='valid@email.com', password='pw123123')
homepage/tests/followedUserCourses_model_tests.py:13:        app_user1 = AppUser.create_app_user('user1', 'user2@gmail.com', '1234')

please check it.

**Review by kobihk — CHANGES_REQUESTED** [review]

**Review by VeraTs — APPROVED** [review]

File: homepage/forms.py (hunk 2 above)
Comment (LeonRabinovich) [reply homepage/forms.py]: Changed

File: homepage/forms.py (hunk 3 above)
Comment (LeonRabinovich) [reply homepage/forms.py:41]: Fixed all occurrences

**Review by kobihk — APPROVED** [review]

**Pull request merged**